src/schemathesis/cli/output.py

```diff
@@ -62,3 +69,65 @@ def pretty_print_stats(stats: runner.StatsCollector, hypothesis_output: Optional
                     click.style(verdict, fg=color, bold=True),
                 )
             )
+
+
+def percentage(position: int, length: int) -> str:
+    return f"[{position * 100 // length}%]"
+
+
+def pretty_print_test_progress(results_generator: Generator[events.ExecutionEvent, None, None]) -> StatsCollector:
+    # Starting progress position
+    position = 0
+    errors = 0
+    # Terminal size and fields position
+    columns = shutil.get_terminal_size((100, 50)).columns
+    col1_len = 8  # Method column
+    col2_len = 10  # Test endpoint column
+    col3_len = 4  # Test result column
+    col4_len = 10  # Progress column
+    # Initial runner state
+    init = next(results_generator)
+    print_header(init.schema.endpoints_count)  # type: ignore
+    for result in results_generator:
+        # Print test method and endpoint before test execution
+        if isinstance(result, events.BeforeExecution):
+            # Collect errors count before test execution
+            errors = sum([check["error"] for check in result.statistic.data.values()])
+            # Re-calculate fields position
+            col2_len = len(result.endpoint.path)
+            template = f"    {{:<{col1_len}}} {{:<{col2_len}}} "
+            click.echo(template.format(result.endpoint.method, result.endpoint.path), nl=False)
+        # Print test result and progress after execution
+        if isinstance(result, events.AfterExecution):
+            position += 1
+            # Recalculate number of errors to obtain test state after execution
+            new_errors = sum([check["error"] for check in result.statistic.data.values()])
+            template = f"{{:{col3_len}}} "
+            if new_errors > errors:
+                click.echo(template.format("F"), nl=False)
+            else:
+                click.echo(template.format("."), nl=False)
+            col4_len = int(columns) - col1_len - col2_len - col3_len - 10
+            click.echo(f"{{:>{col4_len}}}".format(percentage(position, init.schema.endpoints_count)))  # type: ignore
+        if isinstance(result, events.Finished):
+            return result.statistic
+    return StatsCollector()
```

Comment:
I see no point to return an empty StatsCollector. Probably the return type should be `None`

Reply:
Hm, please, ignore the comment above